test/testUcumLhcUtils.spec.js

```diff
@@ -222,68 +221,404 @@ it("should return a message for invalid unit strings", function() {
+    assert.equal(resp.toUnit, undefined, resp.toUnit);
+  });
+
+  // Ignore excess optional parameters if they are not needed for the conversion
+  // Do not raise error if charge is provided despite neither unit being an eq unit
+  // e.g. converting between mol and g
+  it("should return 40.08g for a request to convert 1 mol to g with charge 2 and mw 40.08", function() {
+    var resp = utils.convertUnitTo('mol', 1, 'g', {
+      charge: 2,
+      molecularWeight: 40.08
+    });
+    assert.equal(resp.status, 'succeeded', resp.status + resp.msg);
+    assert.equal(resp.toVal.toPrecision(4), 40.08, resp.toVal.toPrecision(4));
+  });
+
   it("should return 96 for a request to convert 5.33 mmol/L to mg{molybdenum}/dL weight 180.156", function() {
-    var resp5 = utils.convertUnitTo('mmol/L', 5.33, 'mg{molybdenum}/dL', false, 180.156);
+    var resp5 = utils.convertUnitTo('mmol/L', 5.33, 'mg{molybdenum}/dL', { molecularWeight: 180.156 });
     assert.equal(resp5.status, 'succeeded', resp5.status + resp5.msg);
     assert.equal(resp5.toVal.toPrecision(2), 96, resp5.toVal);
   });
 
   it("should return 5.33 for a request to convert 96 mg/dL to mmol/L weight 180.156", function() {
-    var resp5 = utils.convertUnitTo('mg/dL', 96, 'mmol/L', false, 180.156);
+    var resp5 = utils.convertUnitTo('mg/dL', 96, 'mmol/L', { molecularWeight: 180.156 });
     assert.equal(resp5.status, 'succeeded', resp5.status + resp5.msg);
     assert.equal(resp5.toVal.toPrecision(3), 5.33, resp5.toVal.toPrecision(3));
   });
 
+  it("should return 0.533 for a request to convert 96 g to mol with mw 180.156", function() {
+    var resp = utils.convertUnitTo('g', 96, 'mol', { molecularWeight: 180.156 });
+    assert.equal(resp.status, 'succeeded', resp.status + resp.msg);
+    assert.equal(resp.toVal.toPrecision(3), 0.533, resp.toVal.toPrecision(3));
+  });
+
+  it("should return 96 g for a request to convert 0.533 mol with mw 180.156", function() {
+    var resp = utils.convertUnitTo('mol', 0.533, 'g', { molecularWeight: 180.156 });
+    assert.equal(resp.status, 'succeeded', resp.status + resp.msg);
+    assert.equal(resp.toVal.toPrecision(2), 96, resp.toVal.toPrecision(2));
+  });
+
+  /** Mol to eq conversion tests
+   * equivalents = moles * charge
+   */
+
+  // Should throw an error if no charge is provided for a conversion to eq since charge is needed for the calculation
+  it("should return error for a request to convert mol to eq with no charge and mw 40.08", function() {
+    var resp = utils.convertUnitTo('mol', 1, 'eq');
+    assert.equal(resp.status, 'failed', resp.status + resp.msg);
+    assert.equal(resp.msg[0], Ucum.needEqChargeMsg_,
+      resp.msg[0]);
+    assert.equal(resp.toVal, null, resp.toVal);
+    assert.equal(resp.fromUnit, undefined, resp.fromUnit);
+    assert.equal(resp.toUnit, undefined, resp.toUnit);
+  });
+
+  it("should return 1 for a request to convert 1 mol/L to eq/L with charge 1", function() {
```

Comment:
I think there is some redundancy in these five tests.  Maybe just keep one that returns an integer and one that returns a decimal number?